### PR TITLE
Wiki pages with 0 likes shown on /wiki/liked

### LINKED ISSUE #2310: Wiki pages with 0 likes not shown on '/wiki/liked' page (sorted by no. of likes)
Hi, this is a first-timers-only issue. This means we've worked to make it more legible to folks who either haven't contributed to our codebase before or even folks who haven't contributed to open source before.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you :gift_heart:

If you have contributed before, consider leaving this one for someone new, and looking through our general help wanted issues. Thanks!
:thinking: What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.
### Please describe the problem

Wiki page when sorted according to likes does not show pages with 0 likes:-
liked-2
liked-3

### Solution

We need to refactor the query given below to select pages with 0 likes too :- 
https://github.com/publiclab/plots2/blob/57f0d6f97844b9ba070f7a048fc9a942c1ecf0c8/app/controllers/wiki_controller.rb#L337-L339
The above could be replaced by:-
`
    @wikis = Node.limit(40)
      .order('node.cached_likes DESC')
      .where("status = 1 AND nid != 259 AND (type = 'page' OR type = 'tool' OR type = 'place') AND 
                   cached_likes >= 0")
`

:clipboard: Step by Step

:raising_hand_woman: Claim this issue: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

:floppy_disk: Commit your changes

:twisted_rightwards_arrows: Start a Pull Request.

:checkered_flag: Done Ask in comments for a review :)

:thinking::question: Questions?

Leave a comment below!

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. If someone seems stuck, offer them some help! Otherwise, take a look at some other issues you can help with. Thanks!
Thank you!

Your help makes Public Lab better! We deeply appreciate your helping refine and improve this site.

### PR DESCRIPTION
**Pull request by jsmayo**
Modifies code to select and include wiki pages with 0 likes. 
Fixes #2310 

p.s. Sorry for the prolonged PR, I was out of town and apparently did not initiate the request in the right spot! My apologies!

### CODE REVIEW

File: app/controllers/wiki_controller.rb [hunk 1]
```diff
@@ -335,8 +335,9 @@ def popular
   def liked
     @title = I18n.t('wiki_controller.well_liked_wiki_pages')
     @wikis = Node.limit(40)
-      .order('node.cached_likes DESC')
-      .where("status = 1 AND nid != 259 AND (type = 'page' OR type = 'tool' OR type = 'place') AND cached_likes > 0")
+      .order('node.cached_likes DESC') 
+      .where("status = 1 AND nid != 259 AND (type = 'page' OR type = 'tool' OR type = 'place') AND cached_likes >= 0")
+
```
[the remark this answers is not in the document]
Comment (jsmayo) [reply app/controllers/wiki_controller.rb]: I went with the styling that I thought was already prevalent in the src code.

[the remark this answers is not in the document]
Comment (seafr) [reply app/controllers/wiki_controller.rb]: I see your point @jsmayo. This seems to be the styling prevalent in some of the methods in that particular controller. The code is correct though and the PR will be merged soon. Thanks!
What do you think, @ryzokuken?

[the remark this answers is not in the document]
Comment (ryzokuken) [reply app/controllers/wiki_controller.rb]: Yeah, I didn't have any problem in particular with the newline, I was just wondering if it served any purpose. If it makes things easier to read or more consistent, then keep it by all means.

**Comment by jywarren** [comment]
I just updated the title to be a bit more descriptive but this looks perfect otherwise! Thank you so much! 👍 👍 🎉

**Comment by jsmayo** [comment]
Yay! Thank you as well, for both the patience and opportunity for learning how to do this properly :D

**Pull request merged**